mirr254/Bright-Events
# Ft database integration 153052558

**What does this PR do?**
Migrates the application from non-persistent to persistent data storage using postgresql

**Description of the tasks completed**
- Implements token based authentication
- Secures events endpoints. Only authenticated users with valid token can operate on secured endpoints
- Moves the application from non-persistent to persistent using postgresql 

**How should the PR be tested**

Clone the repo

- `cd` to the the directory and install the requirements `pip install requirements.txt`
- using git, checkout to the branch `Ft database integration 153052558`
- Run the migration script to create database required
`python manage.py db init
python manage.py db migrate
python manage.py db upgrade`

Run the app
  `python run.py`

All the endpoints have been affected by the PR. And they access persistent data from the postgresql. The endpoints are:
    

-     POST    /api/auth/register/
-     POST    /api/auth/login/
-     POST    /api/auth/logout/
-     POST    /api/auth/reset-password/<string:email>

When accessing the events edpoints, include the token got from login endpoint in request header as shown `x-access-token : <token>` 
-     POST    /api/events/
-     GET     /api/events/
-     GET     /api/events/<int:eventid>/
-     PUT     /api/events/<int:eventid>/
-     POST    /api/events/<int:eventid>/rsvp
-     GET    /api/events/<int:eventid>/rsvp
Make sure all the tests are running by executing the command

  `nosetests tests/`

**Related Pivotal stories **
[153052558](https://www.pivotaltracker.com/story/show/153052558)
[153052401](https://www.pivotaltracker.com/story/show/153052401)
